Sources/PackageModelSyntax/AddTarget.swift

```diff
@@ -234,10 +234,10 @@ public struct AddTarget {
         case .macro:
             """
             \(imports)
-            struct \(raw: target.name): Macro {
```

Comment:
How about using `c99name` instead?

Reply:
That's a very good point. `c99name` would certainly cover most edge cases, such as using dash-case, special characters, and digits in the first characters. Well, it's already used for language-level target name sanitizing so it must be robust enough for this use case. 

The only remaining concern is that `c99` (obviously) doesn't cover Swift's reserved words. However, in that case, we could just escape the reserved word based on a list taken e.g. from [swift-syntax](https://github.com/swiftlang/swift-syntax/blob/main/Sources/SwiftSyntax/generated/Keyword.swift), which `SwiftPM` already depends on.

I believe combining these two safeguards should be a good enough way to proceed. What do you think?

Reply:
I'm just visiting. :) There may be a better way to avoid reserved names—the package owners probably have ideas!

Reply:
Alright :) @DougGregor, maybe you have some ideas about this?

Reply:
Both of your ideas together (`cc9name` and using the swift-syntax keyword checking) sound reasonable to me.